src/Colors/index.js

```diff
@@ -0,0 +1,101 @@
+import { mergeColors, calculateAlpha, isObject, isString } from './utils.js'
+
+let colors = {
+  white: '#ffffff',
+  black: '#000000',
+  red: '#ff0000',
+  green: '#00ff00',
+  blue: '#0000ff',
+  yellow: '#feff00',
+  cyan: '#00feff',
+  magenta: '#ff00ff',
+}
+
+const normalizedColors = {
+  //store for normalized colors
+}
+
+const calculateColor = (value, options) => {
+  let targetColor = normalizeColor(value)
+  if (!isNaN(options)) {
+    options = { alpha: options }
+  }
+  if (options && isObject(options)) {
+    if (!isNaN(options.darken)) {
+      targetColor = mergeColors(targetColor, 0xff000000, 1 - options.darken)
+    }
+
+    if (!isNaN(options.lighten)) {
+      targetColor = mergeColors(targetColor, 0xffffffff, 1 - options.lighten)
+    }
+
+    if (!isNaN(options.opacity)) {
+      options.alpha = options.opacity / 100
+    }
+
+    if (!isNaN(options.alpha)) {
+      targetColor = calculateAlpha(targetColor, options.alpha)
+    }
+  }
+  return targetColor || 0
+}
+
+const normalizeColor = color => {
+  let targetColor = colors[color]
+  if (!targetColor) {
+    targetColor = color
+  }
+  if (isString(targetColor) && /^#[0-9A-F]{6}$/i.test(targetColor.toUpperCase())) {
```

Comment:
- I don't think we need `toUppperCase` since the regexp is already case insenstive
- maybe we should add support for `#333` as well (since it's so common in CSS, I guess it's very possible people will use the hex shorthand :) )

Reply:
Reworked this part of the function, which removed the toUppercase and added support for 3 digit and converting it to 6 digit.